src/core/webidl.js

```diff
@@ -5,181 +5,113 @@
-      const className = parent ? "idlName" : "idlID";
-      if (data.type === "enum-value") {
-        return idlAnchor;
-      }
-      return hyperHTML`<span class="${className}">${idlAnchor}</span>`;
-    },
-    type: contents => hyperHTML`<span class="idlType">${contents}</span>`,
-    inheritance: contents =>
-      hyperHTML`<span class="idlSuperclass">${contents}</span>`,
-    definition: createIdlElement,
-    extendedAttribute: contents =>
-      hyperHTML`<span class="extAttr">${contents}</span>`,
-    extendedAttributeReference(name) {
-      if (!extendedAttributesLinks.has(name)) {
-        return hyperHTML`<a>${name}</a>`;
-      }
-      return hyperHTML`<a data-cite="${extendedAttributesLinks.get(
-        name
-      )}">${name}</a>`;
-    },
-  };
-  const result = webidl2.write(parse, { templates });
-  return hyperHTML`<pre class="def idl">${result}</pre>`;
-}
+const templates = {
+  wrap(items) {
+    return items
+      .reduce(flatten, [])
+      .filter(x => x !== "")
+      .map(x => (typeof x === "string" ? new Text(x) : x));
+  },
+  trivia(t) {
+    if (!t.trim()) {
+      return t;
+    }
+    return hyperHTML`<span class='idlSectionComment'>${t}</span>`;
+  },
+  generic(keyword) {
+    // Shepherd classifies "interfaces" as starting with capital letters,
+    // like Promise, FrozenArray, etc.
+    return /^[A-Z]/.test(keyword)
+      ? hyperHTML`<a data-xref-type="interface">${keyword}</a>`
+      : // Other keywords like sequence, maplike, etc...
+        hyperHTML`<a data-xref-type="dfn">${keyword}</a>`;
+  },
+  reference(wrapped) {
+    return hyperHTML`<a data-xref-type="_IDL_">${wrapped}</a>`;
+  },
+  name(escaped, { data, parent }) {
+    if (data.idlType && data.idlType.type === "argument-type") {
+      return hyperHTML`<span class="idlParamName">${escaped}</span>`;
+    }
+    const idlAnchor = createIdlAnchor(escaped, data, parent);
+    const className = parent ? "idlName" : "idlID";
+    if (data.type === "enum-value") {
+      return idlAnchor;
+    }
+    return hyperHTML`<span class="${className}">${idlAnchor}</span>`;
+  },
+  type(contents) {
+    return hyperHTML`<span class="idlType">${contents}</span>`;
+  },
+  inheritance(contents) {
+    return hyperHTML`<span class="idlSuperclass">${contents}</span>`;
+  },
+  definition(contents, { data, parent }) {
+    const className = getIdlDefinitionClassName(data);
+    switch (data.type) {
+      case "includes":
+      case "enum-value":
+        return hyperHTML`<span class='${className}'>${contents}</span>`;
+    }
+    const parentName = parent ? parent.name : "";
+    const { name, idlId } = getNameAndId(data, parentName);
+    return hyperHTML`<span class='${className}' id='${idlId}' data-idl data-title='${name}'>${contents}</span>`;
+  },
+  extendedAttribute(contents) {
+    const result = hyperHTML`<span class="extAttr">${contents}</span>`;
+    return result;
+  },
+  extendedAttributeReference(name) {
```

Comment:
@saschanaz, I've got a bit of a problem here... given:

```
[Exposed=Window, Worker]
```

It turns out `Worker` is not actually defined by any specification. And `Window` is also a bit dubious (probably why BikeShed doesn't link these either). I wonder if we can skip passing those as referenced in WebIDL2.js?

Reply:
Actually, Window is fine... it has [Global]. Let me see if I can track down Worker.

Reply:
It's unclear to me if `Worker` maps to `WorkerGlobalScope`... it feels like it should, but I can't seem to put all the pieces together looks at the various specs.

Reply:
Argh... hmmm... maybe we should avoid linking anything after "=" for all WebIDL things.

Reply:
Typically adding `GlobalScope` postfix should be enough. `Window` is an exception.

Reply:
ah, ok. We can try that.

Reply:
Linking the workers ended up being ugly and causing issues elsewhere :( Will need to think about it more. 

I'd be happy with not linking for now tho, as it addresses the main problems.